### PR TITLE
[Paddle-TRT]Support argsort by TopK

### PR DESCRIPTION
**Pull request by ckl117**
### PR Category

Inference

### PR Types

Others

### Description

Pcard-71500
When the number of items to be sorted is less than or equal to 3840, use ITopKLayer to implement argsort.

### CODE REVIEW

File: paddle/fluid/inference/tensorrt/convert/argsort_op.cc [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    if (need_cast) {
+      auto* cast_layer1 =
+          TRT_ENGINE_ADD_LAYER(engine_, Identity, *input_tensor);
+      cast_layer1->setOutputType(0, nvinfer1::DataType::kFLOAT);
+      cast_layer1->getOutput(0)->setType(nvinfer1::DataType::kFLOAT);
+      input_tensor = cast_layer1->getOutput(0);
+    }
+
+    auto* layer = TRT_ENGINE_ADD_LAYER(
+        engine_, TopK, *input_tensor, operation, 0, 1 << axis);
```
Comment (ming1753) [inline paddle/fluid/inference/tensorrt/convert/argsort_op.cc]: k的默认值设置为1

Comment (ming1753) [reply paddle/fluid/inference/tensorrt/convert/argsort_op.cc:62]: 先合入了，后续pr需要修复这个问题

File: paddle/fluid/inference/tensorrt/op_teller.cc [hunk 2]
```diff
@@ -2778,6 +2778,30 @@ struct SimpleOpTypeSetTeller : public Teller {
...
+        VLOG(3) << op_type << " needs attributes: descending and axis.";
+      }
+      auto x_var_name = desc.Input("X")[0];
+      auto* x_var_desc = block->FindVarRecursive(x_var_name);
+      std::vector<int64_t> shape = x_var_desc->GetShape();
+      int axis = PADDLE_GET_CONST(int, desc.GetAttr("axis"));
+      if (axis < 0) {
+        axis += shape.size();
+      }
+      if (shape[axis] > 3840) {
```
Comment (ming1753) [inline paddle/fluid/inference/tensorrt/op_teller.cc]: 动态shape这里只能获取到-1，建议不进trt

File: paddle/fluid/inference/tensorrt/convert/argsort_op.cc (hunk 1 above)
Comment (ckl117) [reply paddle/fluid/inference/tensorrt/convert/argsort_op.cc]: ok

File: paddle/fluid/inference/tensorrt/op_teller.cc (hunk 2 above)
Comment (ckl117) [reply paddle/fluid/inference/tensorrt/op_teller.cc]: done

**Review by ming1753 — APPROVED** [review]

**Pull request merged**